spec/lib/simplekiq/monitored_worker_spec.rb

```diff
@@ -0,0 +1,91 @@
+require 'spec_helper'
+require 'simplekiq/testing'
+
+RSpec.describe Simplekiq::MonitoredWorker do
+  class self::MonitoredTestWorker
+    include Simplekiq::MonitoredWorker
+
+    def perform(args)
+      check_args(args)
+    end
+
+    def check_args(args); end
+  end
+
+  let(:fake_params) { { id: 1 } }
+  let(:job_count) { 100 }
+  let(:job_id) { SecureRandom.hex }
+  let(:monitoring_params) { { enqueuer_job_id: job_id } }
+  let(:worker_class) { self.class::MonitoredTestWorker }
+
+  before do
+    Thread.current['sidekiq.enqueuer_job_id'] = job_id
+    Thread.current['sidekiq.enqueued_jobs_count'] = 100
+  end
+
+  shared_examples 'adds enqueuer job id to sidekiq and increments job count' do
+    it 'adds enqueuer job id to params' do
+      expect_any_instance_of(MonitorProgress)
+        .to receive(:perform).with(fake_params.merge(monitoring_params))
+      async_perform
+    end
+
+    it 'increments the enqueued jobs count' do
+      async_perform
+      expect(Thread.current['sidekiq.enqueued_jobs_count']).to eq 101
+    end
+  end
+
+  describe '.perform_async' do
+    subject(:async_perform) do
+      Sidekiq::Testing.inline! { worker_class.perform_async(fake_params) }
+    end
+
+    include_examples 'adds enqueuer job id to sidekiq and increments job count'
+  end
+
+  describe '.perform_in' do
+    subject(:async_perform) do
+      Sidekiq::Testing.inline! { worker_class.perform_in(10, fake_params) }
+    end
+
+    include_examples 'adds enqueuer job id to sidekiq and increments job count'
+  end
+
+  describe '.perform_at' do
+    subject(:async_perform) do
+      Sidekiq::Testing.inline! { worker_class.perform_at(Time.now, fake_params) }
+    end
+
+    include_examples 'adds enqueuer job id to sidekiq and increments job count'
+  end
+
+  describe '#perform' do
```

Comment:
your subject calls perform_async not worker_class.new.perform. name the describe block accordingly.
```suggestion
  describe '.perform_async' do
```

Reply:
Changed it to actually call `perform`